debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/AbstractMysqlFieldReader.java

```diff
@@ -49,10 +60,14 @@ else if (column.jdbcType() == Types.TINYINT || column.jdbcType() == Types.SMALLI
         // DBZ-2673
         // It is necessary to check the type names as types like ENUM and SET are
         // also reported as JDBC type char
-        else if ("CHAR".equals(column.typeName()) ||
-                "VARCHAR".equals(column.typeName()) ||
-                "TEXT".equals(column.typeName())) {
-            return rs.getBytes(columnIndex);
+        else if (!connectorConfig.customConverterRegistry().isEmpty() && TEXT_DATATYPES.contains(column.typeName())) {
+            try {
+                return rs.getString(columnIndex).getBytes(column.charsetName());
```

Comment:
may throws npe here, since rs.getString() may return nuil

Reply:
Mh, yeah, good spot. Logged https://issues.redhat.com/browse/DBZ-4931 for this.